superset/migrations/versions/fc3a3a8ff221_migrate_filter_sets_to_new_format.py

```diff
@@ -0,0 +1,233 @@
+        filter_obj["filterState"] = current_state
+
+        # create new extraFormData field
+        old_extra_form_data = filter_obj.pop("extraFormData", {})
+        extra_form_data = {}
+        filter_obj["extraFormData"] = extra_form_data
+
+        # upgrade append filters
+        appends = old_extra_form_data.pop("append_form_data", {})
+        extra_form_data.update(appends)
+
+        # upgrade override filters
+        overrides = old_extra_form_data.pop("override_form_data", {})
+        for override_key, override_value in overrides.items():
+            # nested extras are also moved up to main object
+            if override_key == "extras":
+                for extra_key, extra_value in override_value.items():
+                    extra_form_data[extra_key] = extra_value
+            else:
+                extra_form_data[override_key] = override_value
+    return changed_filters
+
+
+def downgrade_filter_set(filter_set: Dict[str, Any]) -> int:
+    changed_filters = 0
+    old_data_mask = filter_set.pop("dataMask", {})
+    native_filters = {}
+    data_mask = {"nativeFilters": native_filters}
+    filter_set["dataMask"] = data_mask
+    for filter_id, filter_obj in old_data_mask.items():
+        changed_filters += 1
+        # move filter object down one level
+        native_filters[filter_id] = filter_obj
+
+        # downgrade filter state
+        filter_state = filter_obj.pop("filterState", {})
+        filter_obj["currentState"] = filter_state
+
+        old_extra_form_data = filter_obj.pop("extraFormData", {})
+        extra_form_data = {}
+        filter_obj["extraFormData"] = extra_form_data
+
+        # downgrade append keys
+        append_form_data = {}
+        extra_form_data["append_form_data"] = append_form_data
+        for key in EXTRA_FORM_DATA_APPEND_KEYS:
+            value = old_extra_form_data.pop(key, None)
+            if value is not None:
+                append_form_data[key] = value
+        if not append_form_data:
+            del extra_form_data["append_form_data"]
+
+        # downgrade override keys
+        override_form_data = {}
+        extra_form_data["override_form_data"] = override_form_data
+        for key in EXTRA_FORM_DATA_OVERRIDE_KEYS:
+            value = old_extra_form_data.pop(key, None)
+            if key in EXTRA_FORM_DATA_OVERRIDE_EXTRA_KEYS:
+                extras = override_form_data.get("extras", {})
+                extras[key] = value
+            elif value is not None:
+                override_form_data[key] = value
+        if not override_form_data:
+            del extra_form_data["override_form_data"]
+
+    return changed_filters
+
+
+def upgrade():
+    bind = op.get_bind()
+    session = db.Session(bind=bind)
+
+    dashboards = (
+        session.query(Dashboard)
+        .filter(Dashboard.json_metadata.like('%"filter_sets_configuration"%'))
+        .all()
+    )
+    changed_filter_sets, changed_filters = 0, 0
+    for dashboard in dashboards:
```

Comment:
```suggestion
    dashboards = (
        session.query(Dashboard)
        .filter(Dashboard.json_metadata.like('%"filter_sets_configuration"%'))
    )
    changed_filter_sets, changed_filters = 0, 0
    for dashboard in dashboards.yield_per(100):
```

Reply:
Thanks - this is a great pattern! As this is behind a FF, I don't expect we'll have lots of these dashboards. Also in the interest of CI time and unblocking other PRs that are dependent on this I think I'll just merge now, but I can open up a follow up PR to implement this.